server/socket.js

```diff
@@ -38,13 +38,8 @@ function saveEventToDb(eventType) {
 				return;
 			}
 
-			const eventInDb = new Event();
-			eventInDb._id = extract(data, 'id');
-			eventInDb.type = eventType;
-			eventInDb.isoTimestampSent = extract(data, 'timestamp');
+			const eventInDb = new Event(data);
 			eventInDb.isoTimestampReceived = isoTimestampReceived;
```

Comment:
```suggestion
			const eventInDb = new Event({
				...data,
				isoTimestampReceived,
			});
```

Reply:
Great idea. Bonus point if `Event` can be immutable :)

Reply:
Interesting to know that `const` in JavaScript allows properties to be modified. 

We cannot make `Event` immutable since mongoose updates the model object and add `$errors` property to it.